Framework/DataHandling/src/SaveTBL.cpp

```diff
@@ -83,16 +83,28 @@ void SaveTBL::exec() {
         std::string groupHeading = columnHeadings[columnIndex];
         if (ws->getColumn(groupHeading)->type() != "int") {
           file.close();
-          remove(filename.c_str());
-          throw std::runtime_error(groupHeading +
-                                   " Column must be of type \"int\"");
+          int error = remove(filename.c_str());
+          if (error == 0)
+            throw std::runtime_error(groupHeading + " column must be of type "
+                                                    "\"int\". The TBL file "
+                                                    "will not be saved.");
+          else
+            throw std::runtime_error(
```

Comment:
Yes, I was messing around with different error messages for both cases and I didn't really know how best to phrase it, but in the event that remove has failed I agree and I think we should add something extra about remove failing.